src/domain/checkForAndPlaceOrder.ts

```diff
@@ -103,159 +102,267 @@ export async function checkForAndPlaceOrder(
+      // Stop the timer
+      ownerTimer();
 
-      const pollResult = await _processConditionalOrder(
+      return {
         owner,
-        conditionalOrder,
-        blockTimestamp,
-        blockNumber,
-        context,
-        orderRef
-      );
+        results,
+      };
+    }
+  );
 
-      // Don't try again the same order, in case that's the poll result
-      if (pollResult.result === PollResultCode.DONT_TRY_AGAIN) {
-        ordersPendingDelete.push(conditionalOrder);
-      }
+  // Get all the results
+  const ownerResults = await Promise.all(ownerPromises);
 
-      // Save the latest poll result
-      conditionalOrder.pollResult = {
-        lastExecutionTimestamp: blockTimestamp,
-        blockNumber: blockNumber,
+  // Stop the timer
+  blockTimer();
 
-        result: pollResult,
-      };
+  // Start the post-processing timer
+  const postProcessingTimer = pollingPostProcessingDurationSeconds
+    .labels(chainId.toString(), blockNumber.toString())
+    .startTimer();
 
-      // Log the result
-      const unexpectedError =
-        pollResult?.result === PollResultCode.UNEXPECTED_ERROR;
+  // Now that we have all the results, we can update the registry synchronously
+  let hasErrors = false;
 
-      // Print the polling result
-      const isError = pollResult.result !== PollResultCode.SUCCESS;
-      const resultDescription =
-        pollResult.result +
-        (isError && pollResult.reason ? `. Reason: ${pollResult.reason}` : "");
+  // Process all the orders. We do this in a try/catch so that we can apply the
+  // post-processing timer.
+  try {
+    for (const { owner, results } of ownerResults) {
+      const conditionalOrders = ownerOrders.get(owner);
 
-      log[unexpectedError ? "error" : "info"](
-        `Check conditional order result: ${getEmojiByPollResult(
-          pollResult?.result
-        )} ${resultDescription}`
-      );
-      if (unexpectedError) {
-        log.error(`UNEXPECTED_ERROR Details:`, pollResult.error);
+      if (conditionalOrders === undefined && results.length > 0) {
+        throw new Error(
+          "Unexpected error: conditionalOrders is undefined but results is not empty"
+        );
+      } else if (conditionalOrders) {
+        // Process all the orders
+        for (const result of results) {
+          if (!result) {
+            throw new Error("Unexpected error: orderResult is undefined");
+          }
+
+          const { order, _delete, unexpectedError } = result;
+
+          if (unexpectedError) {
+            hasErrors = true;
+          }
+
+          // First calculate the `conditionalOrderId` from the `ConditionalOrder` params
+          const id = ConditionalOrderSDK.leafToId(order.params);
+
+          // Search for the order in the registry and update / delete it
+          for (const o of Array.from(conditionalOrders.values())) {
+            if (ConditionalOrderSDK.leafToId(o.params) === id) {
```

Comment:
Arent you calculating the leafToId for all `conditionalOrders.values()` orders for each result you have? 
it looks like you repeat the work over and over.

Isn't it better to build a Map "ID --> Order"?